clients/drcachesim/reader/v2p_reader.cpp

```diff
@@ -0,0 +1,116 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, LLC  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, LLC nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, LLC OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "v2p_reader.h"
+
+#include <cstdint>
+#include <stdint.h>
+
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <unordered_map>
+
+#include "utils.h"
+
+namespace dynamorio {
+namespace drmemtrace {
+
+v2p_reader_t::v2p_reader_t()
+{
+    /* Empty. */
+}
+
+bool
+v2p_reader_t::gen_v2p_map(std::string path_to_file,
+                          std::unordered_map<uint64_t, uint64_t> &v2p_map)
+{
+    std::ifstream file(path_to_file);
+    if (!file.is_open()) {
+        ERRMSG("ERROR: Failed to open %s\n", path_to_file.c_str());
+        return false;
+    }
+
+    constexpr char virtual_address_key[] = "virtual_address";
```

Comment:
Add the example entry syntax from the PR description in a comment?  I had to keep referring back to it while reading this code.

Reply:
Done. https://github.com/DynamoRIO/dynamorio/pull/6926#discussion_r1733933698